Align VPN with Monorepo develop

durch opened the pull request:
Bring us back to monorepo develop!

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)
<!-- Reviewable:end -->

durch committed:
Pave over erorr flattening, errors in general need rework

durch committed:
Use proc macro to configure the probe

benedettadavico committed:
Update main.rs

neacsu committed:
Include ports fix

benedettadavico committed:
update timeout to 180 sconds

benedettadavico committed:
bump timeout to 300s

benedettadavico committed:
Update lib.rs

benedettadavico committed:
Update lib.rs

benedettadavico committed:
Update Cargo.toml

jstuczyn committed:
api client fixes

durch committed:
Use latest client-registry

durch committed:
TaskManager and other issues

durch committed:
No taskclient for mixnet

durch committed:
Pin to develop

durch committed:
fmt

simonwicky commented on nym-vpn-core/crates/nym-vpn-account-controller/src/state_machine/syncing_state/mod.rs:
```diff
@@ -156,24 +156,18 @@ impl SyncingState {
-                        tracing::error!("Error trying to get account summary : {e}");
-                        Err(SyncError::ApiRequestError)
-                    }
+                let error_response = NymErrorResponse::from(e);
```
This is a behavior change since we were using `NymErrorResponse::try_from(e)` to distinguish between "Request was fine, API returned an error" and "Something wrong happened, the API probably didn't get our request"
The latter lead to retries, the former not

simonwicky replied:
That explains why there is now an unused variant in `SyncError` later in the file

durch replied:
Yeah, I've added TODOs (I hope)

pronebird replied:
Would it not be better to keep the original approach with `TryFrom`?
<!-- Reviewable comment -O_WpiT--h1vmWOOKa0n:bdmok7f -->
<!-- Sent from Reviewable.io -->

durch committed:
Cleanup deprecated

durch committed:
Cleanup other platforms

pronebird reviewed (commented):
> @pronebird reviewed 2 of 36 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 2 of 37 files reviewed, 1 unresolved discussion (waiting on @mmsinclair, @neacsu, and @simonwicky)



<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> @pronebird reviewed 13 of 36 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 15 of 37 files reviewed, 2 unresolved discussions (waiting on @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-account-controller/src/commands/handler.rs` line 146 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S4I6v3cy7x5BREnyM:-O_S4I6v3cy7x5BREnyN:b3b0vlf) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-account-controller/src/commands/handler.rs#L146)):*
> ```XML
>         .await
>     {
>         #[allow(clippy::unnecessary_fallible_conversions)]
> ```

This does not look right. This needs to be fixed.


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 15 of 37 files reviewed, 3 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-account-controller/src/state_machine/syncing_state/mod.rs` line 315 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S5U8C0rls8T1sgpGC:-O_S5U8C0rls8T1sgpGD:b-kedkg2) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-account-controller/src/state_machine/syncing_state/mod.rs#L315)):*
> ```XML
> impl From<VpnApiClientError> for SyncError {
>     fn from(value: VpnApiClientError) -> Self {
>         let error_response = NymErrorResponse::from(value);
> ```

I'd assume that not every error is convertible to `NymErrorResponse` and hence I'd expect to have a method to query this, i.e:

```rs
impl VpnApiClientError {
  pub fn nym_error_response(&self) -> Option<NymErrorResponse> { .. }
}
```

Infallible `From` in this case just does not make sense to me.


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> @pronebird reviewed 1 of 36 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 4 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 60 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S6SwL2nok6W9DPmFF:-O_S6SwL2nok6W9DPmFG:b-43s3od) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L60)):*
> ```XML
>     ) -> Result<Self> {
>         // Get VPN API URLs from network details
>         #[allow(deprecated)]
> ```

What's been deprecated?


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 5 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-account-controller/tests/integrations/paths.rs` line 100 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S6WwLCQpn2tyyImld:-O_S6WwLCQpn2tyyImle:b-g78ext) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-account-controller/tests/integrations/paths.rs#L100)):*
> ```XML
> // TODO: we need error details here, taking them out until we rework the errors
> #[tokio::test]
> #[ignore]
> ```

This does not look right to me.


<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 6 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 589 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S7zs--kPoJJt7FH4I:-O_S7zs0-Np85VPA0ez7:b6eqf7y) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L589)):*
> ```XML
>             Err(err) => {
>                 if let HttpClientError::EndpointFailure {
>                     url: _,
> ```

Nit :maybe use `{ error, .. }` to omit other fields


<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 7 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 522 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S84Cs-pGar1Q86JFC:-O_S84CtBNlIZ2AI-GB7:b6eqf7y) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L522)):*
> ```XML
>             Err(err) => {
>                 if let HttpClientError::EndpointFailure {
>                     url: _,
> ```

Nit :maybe use `{ error, .. }` to omit other fields


<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 8 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 661 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S85iA4SiT0mL7yMD6:-O_S85iA4SiT0mL7yMD7:b6eqf7y) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L661)):*
> ```XML
>             Err(err) => {
>                 if let HttpClientError::EndpointFailure {
>                     url: _,
> ```

Nit :maybe use `{ error, .. }` to omit other fields


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 16 of 37 files reviewed, 9 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/response.rs` line 562 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S99Si7aAt8nFNBqEA:-O_S99Si7aAt8nFNBqEB:bjw8d8c) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/response.rs#L562)):*
> ```XML
> 
> pub fn extract_error_response(err: VpnApiClientError) -> NymErrorResponse {
>     NymErrorResponse {
> ```

Do not do this. You're removing all errors that are being held in `NymErrorResponse`. We need to obtain error response in some way.


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> @pronebird reviewed 1 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 17 of 37 files reviewed, 10 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 348 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_SOEnu9GSUY1lkIiLe:-O_SOEnu9GSUY1lkIiLf:bg991y2) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L348)):*
> ```XML
>             Err(err) => {
>                 if let HttpClientError::EndpointFailure {
>                     url: _,
> ```

Use `, ..` perhaps for fields you don't need


<!-- Sent from Reviewable.io -->

pronebird reviewed (changes requested):
> @pronebird reviewed 11 of 36 files at r2, 3 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 31 of 37 files reviewed, 11 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs` line 165 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_SOxImAPUX3FlQT0E6:-O_SOxImAPUX3FlQT0E7:b7js0t8) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs#L165)):*
> ```XML
>         use nym_vpn_api_client::response::NymErrorResponse;
>         // TODO: Another example of losing information about the original error cause
>         #[allow(clippy::unnecessary_fallible_conversions)]
> ```

This is not acceptable change. Let's focus on correctness vs. compilation errors


<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> @pronebird reviewed 5 of 36 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 36 of 37 files reviewed, 11 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)



<!-- Sent from Reviewable.io -->

durch committed:
Restore error extraction functionality after simplification

The previous error simplification removed generic type parameters from
HttpClientError, which broke the ability to extract structured error
responses from the API. This commit restores that functionality while
keeping the simplified types.

Changes:
- Enhanced extract_error_response to parse JSON from error strings
- Fixed http_client_error() to properly handle Box<HttpClientError>
- Supports both NymErrorResponse and UnexpectedError formats
- Preserves critical fields (code_reference_id, status, message)

This ensures downstream code that checks for specific error conditions
(like FAIR_USAGE_DEPLETED_CODE_ID or "Account not found" messages)
continues to work correctly.

durch committed:
Unignore tests

durch commented:
> > @pronebird reviewed 1 of 36 files at r2.
> > _[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)_ status: 16 of 37 files reviewed, 4 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)
> 
> _[`nym-vpn-core/crates/nym-vpn-api-client/src/client.rs` line 60 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_S6SwL2nok6W9DPmFF:-O_S6SwL2nok6W9DPmFG:b-43s3od) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-api-client/src/client.rs#L60)):_
> 
> > ```
> >     ) -> Result<Self> {
> >         // Get VPN API URLs from network details
> >         #[allow(deprecated)]
> > ```
> 
> What's been deprecated?

TaskManager, and related scaffolding, and GenericError

durch committed:
Fix clippy try_from

durch commented:
> > @pronebird reviewed 11 of 36 files at r2, 3 of 4 files at r3.
> > _[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)_ status: 31 of 37 files reviewed, 11 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)
> 
> _[`nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs` line 165 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_SOxImAPUX3FlQT0E6:-O_SOxImAPUX3FlQT0E7:b7js0t8) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs#L165)):_
> 
> > ```
> >         use nym_vpn_api_client::response::NymErrorResponse;
> >         // TODO: Another example of losing information about the original error cause
> >         #[allow(clippy::unnecessary_fallible_conversions)]
> > ```
> 
> This is not acceptable change. Let's focus on correctness vs. compilation errors

Fix pushed

durch committed:
Address review comments

- Use destructuring pattern { error, .. } for HttpClientError matching
- Simplify error conversion using infallible From trait

durch committed:
fmt

durch committed:
Rebase artifact

durch committed:
Box more errors

pronebird reviewed (changes requested):
> @pronebird reviewed 5 of 20 files at r4.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 24 of 39 files reviewed, 6 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs` line 165 at r3](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_SOxImAPUX3FlQT0E6:-O_WpYnD54Z01lTkd4IN:bxzo1bl) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/db7fc21184029e36a4fd22119224941cbe89c2b2/nym-vpn-core/crates/nym-vpn-lib-types/src/account/mod.rs#L165)):*
<details><summary><i>Previously, durch (Drazen Urch) wrote…</i></summary><blockquote>

Fix pushed
</blockquote></details>

Perhaps clean up the TODO above as it's no longer an example of losing info...
___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/response.rs` line 600 at r4](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_Wovii6w2AI_8d41i2:-O_Wovii6w2AI_8d41i3:b414f4h) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/58ce7116145e6d4aeacbf3ae01197a32de556017/nym-vpn-core/crates/nym-vpn-api-client/src/response.rs#L600)):*
> ```XML
> }
> 
> pub fn extract_error_response(err: VpnApiClientError) -> NymErrorResponse {
> ```

Would it not be better to return`Option<NymErrorResponse>` in case when there is no error response? (not json at all?)


<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> @pronebird reviewed 2 of 20 files at r4.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 26 of 39 files reviewed, 3 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)

___
*[`nym-vpn-core/crates/nym-gateway-directory/src/gateway_client.rs` line 166 at r4](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_Wqje1E6a2VMDTz44r:-O_Wqje1E6a2VMDTz44s:b-pp1ckq) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/58ce7116145e6d4aeacbf3ae01197a32de556017/nym-vpn-core/crates/nym-gateway-directory/src/gateway_client.rs#L166)):*
> ```XML
>         // Use the new unified HTTP client with domain fronting for the main API client
>         let api_client = nym_http_api_client::ClientBuilder::from_network(network_details)
>             .map_err(Box::new)?
> ```

FYI are those types that heavy so everything needs to be Boxed?


<!-- Sent from Reviewable.io -->

durch committed:
Windows clippy

durch commented:
> > @pronebird reviewed 2 of 20 files at r4.
> > _[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)_ status: 26 of 39 files reviewed, 3 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @simonwicky)
> 
> _[`nym-vpn-core/crates/nym-gateway-directory/src/gateway_client.rs` line 166 at r4](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_Wqje1E6a2VMDTz44r:-O_Wqje1E6a2VMDTz44s:b-pp1ckq) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/58ce7116145e6d4aeacbf3ae01197a32de556017/nym-vpn-core/crates/nym-gateway-directory/src/gateway_client.rs#L166)):_
> 
> > ```
> >         // Use the new unified HTTP client with domain fronting for the main API client
> >         let api_client = nym_http_api_client::ClientBuilder::from_network(network_details)
> >             .map_err(Box::new)?
> > ```
> 
> FYI are those types that heavy so everything needs to be Boxed?

Yes, the HttpClientError is the problem, I think we can refactor out of it, but for now we Box

simonwicky reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 26 of 39 files reviewed, 3 unresolved discussions (waiting on @durch, @mmsinclair, @neacsu, and @pronebird)

___
*[`nym-vpn-core/crates/nym-vpn-api-client/src/response.rs` line 600 at r4](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_Wovii6w2AI_8d41i2:-O_kMmjy0ECZHhTKNo-8:byez8dg) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/58ce7116145e6d4aeacbf3ae01197a32de556017/nym-vpn-core/crates/nym-vpn-api-client/src/response.rs#L600)):*
<details><summary><i>Previously, pronebird (Andrej Mihajlov) wrote…</i></summary><blockquote>

Would it not be better to return`Option<NymErrorResponse>` in case when there is no error response? (not json at all?)
</blockquote></details>

To keep the existing code behavior, yes. A `NymErrorResponse` is an error response from the Vpn Api, not any error from the reqwest client. We need to be able to make that distinction


<!-- Sent from Reviewable.io -->

durch committed:
Make NymErrorResponse conversion failable

simonwicky committed:
keep internal error variant for when it's actually internal (#3486)

pronebird reviewed (changes requested):
> @pronebird reviewed 5 of 20 files at r4, 2 of 5 files at r6.
> @pronebird dismissed @simonwicky from a discussion.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 30 of 39 files reviewed, 1 unresolved discussion (waiting on @mmsinclair and @neacsu)

___
*[`nym-vpn-core/crates/nym-vpn-account-controller/src/state_machine/syncing_state/mod.rs` line 204 at r6](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-O_pohVR88CW0FQsActZ:-O_pohVR88CW0FQsAct_:b-2fw609) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/5a89357b3b1a732bcdb76f64afb76b228633b25f/nym-vpn-core/crates/nym-vpn-account-controller/src/state_machine/syncing_state/mod.rs#L204)):*
> ```XML
>                             Err(e) if e.is_retryable() => {
>                                 if self.attempts > MAX_SYNCING_ATTEMPTS {
>                                     tracing::error!("Error trying to get account summary, exhausted retries : {}", e.to_string());
> ```

We should reduce these to debug because we are held accountable for all warns and errors that land in Sentry (via tracing)


<!-- Sent from Reviewable.io -->

durch committed:
Downgrade log spam

pronebird reviewed (commented):
> @pronebird reviewed 5 of 20 files at r4, 1 of 1 files at r5, 3 of 5 files at r6.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: 38 of 39 files reviewed, 1 unresolved discussion (waiting on @durch, @mmsinclair, and @neacsu)



<!-- Sent from Reviewable.io -->

pronebird reviewed (commented):
> @pronebird reviewed 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @mmsinclair and @neacsu)



<!-- Sent from Reviewable.io -->

pronebird reviewed (approved):
> @pronebird reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mmsinclair and @neacsu)



<!-- Sent from Reviewable.io -->

pronebird reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/3375#-:-O_pqeVt0WT_TfTvUBh3:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mmsinclair and @neacsu)



<!-- Sent from Reviewable.io -->
